internal/types/scope/scope.go

```diff
@@ -38,3 +41,19 @@ var Map = map[string]Type{
 	Org.String():     Org,
 	Project.String(): Project,
 }
+
+// AllowedIn returns the set of Scopes a Resource is allowed in.
+func AllowedIn(r resource.Type) []Type {
+	switch r {
+	case resource.Alias, resource.Billing, resource.Worker:
+		return []Type{Global}
+	case resource.Account, resource.AuthMethod, resource.AuthToken, resource.ManagedGroup, resource.Policy, resource.Scope, resource.SessionRecording, resource.StorageBucket, resource.User:
+		return []Type{Global, Org}
+	case resource.All, resource.Group, resource.Role:
```

Comment:
What is `resource.All` used for?

Reply:
`resource.All` corresponds to using the wildcard (`*`) in a grant string, e.g. `ids=hcst_1234567890;type=*;actions=create,read,update`

A couple use cases from our docs: https://developer.hashicorp.com/boundary/docs/configuration/identity-access-management/permission-grant-formats#wildcard-type

Reply:
Ah, of course. Is this function expected to be used with that input anywhere? Does this mapping need to be defined for that resource type?

Reply:
We should never call this function on `resource.All` in `GrantsForUser` (where we'll primarily be using it), so I'm returning an error for the `resource.All` case

Reply:
Thanks, that makes sense!